### PR TITLE
fix: Emit `connectFailed` on connection failure.

### LINKED ISSUE #222: Event for failing reconnection
Hi.

With changes based on this issue: https://github.com/jwalton/node-amqp-connection-manager/issues/145 (I actually commented there but without reply) I am unable to follow reconnection count without introducing my own timer over libs retry time (plus  there is possibility that if target node is unavailable, reconnect timeout will be underlying socket connection timeout, not time from lib options, what creates it more hard to do proper syncing).

I am not against making distinction between disconnect from actually connected "connection" event (new behavior) and not successful reconnection (now missing) event. But introduced changes removed the latter events from the timeline now and it is actually breaking for users who relied on this behavior.

Now my thinking about this is that if library offers the means to reconnect disrupted connection it should also offer the means to monitor this process (it might have some importance for the consumers). We had "connect" and "disconnect" (covering actual disconnect and failed reconnections). Now we have "connect" and "disconnect" (for actual failure of connecting first time or when we get the "close" from amqp) but we miss something like "reconnection-failure".

Whether there should be seperate event or not (payload changes?) is the matter of the discussion but I think there should be events informing about failed reconnections. It is just easier for the library to provide than expect from the users to mimic or approximate. I am open to create PR in free time if we would have some decision how to proceed.

Wdyt?

### PR DESCRIPTION
**Pull request by jwalton**
Fixes #222.

This makes it so instead of emitting a `disconnect` event the first time we fail to connect, AmqpConnectionManager now emits a `connectFailed`.  AmqpConnectionManager emits `connectFailed` on every connection failure, and emits `disconnect` only when we transition from connected to disconnected.

BREAKING CHANGE: We will no longer emit a `disconnect` event on an
initial connection failure - instead we now emit `connectFailed` on each
connection failure, and only emit `disconnect` when we transition from
connected to disconnected.

### CODE REVIEW

File: src/AmqpConnectionManager.ts [hunk 1]
```diff
@@ -143,19 +153,20 @@ export default class AmqpConnectionManager extends EventEmitter implements IAmqp
...
     public reconnectTimeInSeconds: number;
 
+    private _connectionAttempts = 0;
+
+    /**
+     * The number of connection attempts this connection manager has made,
+     * successful, failed, or in-progress..
+     */
+    get connectionAttempts(): number {
+        return this._connectionAttempts;
```
Comment (ssukienn) [inline src/AmqpConnectionManager.ts:167]: Should we maybe rename it to `pendingConnectionAttempts` etc. and reset it on connect? Not sure ever increasing counter has sense?

**Comment by ssukienn** [comment]
Ty a lot for the PR, single comment. Othewise, lgtm.

File: src/AmqpConnectionManager.ts (hunk 1 above)
Comment (jwalton) [reply src/AmqpConnectionManager.ts:167]: `pendingConnectionAttempts` makes it sound like it should either be 1 or 0 (we're trying to connect or we're not).

`connectionAttemptsSinceLastConnect`?  That's a bit wordy.

Comment (jwalton) [reply src/AmqpConnectionManager.ts:167]: I'll take the connection attempt count out for now - you can always track it yourself from the event if you really need it, and then we can get this merged.

Comment (ssukienn) [reply src/AmqpConnectionManager.ts]: That is perfectly fine for me, actually that's how I was doing it till now, so just need to change the listened event. Thank you
